packages/medusa/src/api/admin/campaigns/[id]/route.ts

```diff
@@ -36,6 +36,16 @@ export const POST = async (
   req: AuthenticatedMedusaRequest<AdminUpdateCampaignType & AdditionalData>,
   res: MedusaResponse<HttpTypes.AdminCampaignResponse>
 ) => {
+  const existingCampaign = await refetchCampaign(req.params.id, req.scope, [
```

Comment:
non-blocking suggestion:

These refetch*() methods underneath call the `refetchEntity` method, which calls the remoteQuery.

The find or throw scenario will be standard across all atomic requests, might be worth adding a flag to the fetchEntity method that passes to the remoteQuery's `throwIfKeyNotFound`, which should take care of the error messages as well

Reply:
Had a word for the same with @carlos-r-l-rodrigues this morning. The `throwIfKeyNotFound` doesn't work when the `id` is supplied in the `filters` object. Which seems to be the case with these helpers

Reply:
Yeah thinking not to change too much in all the layers right now! Maybe, we can discuss all this one day and then do the cleanup :)